src/dispatching/dispatcher.rs

```diff
@@ -280,6 +302,92 @@ where
     }
 }
 
+const WORKER_QUEUE_SIZE: usize = 64;
+
+fn spawn_worker<Err>(
+    deps: DependencyMap,
+    handler: Arc<UpdateHandler<Err>>,
+    default_handler: Arc<DefaultHandler>,
+    error_handler: Arc<dyn ErrorHandler<Err> + Send + Sync>,
+) -> WorkerTx
+where
+    Err: Send + Sync + 'static,
+{
+    let (tx, rx) = tokio::sync::mpsc::channel(WORKER_QUEUE_SIZE);
+
+    let deps = Arc::new(deps);
+
+    tokio::spawn(async move {
+        ReceiverStream::new(rx)
+            .for_each(|update| {
+                let deps = Arc::clone(&deps);
+                let handler = Arc::clone(&handler);
+                let default_handler = Arc::clone(&default_handler);
+                let error_handler = Arc::clone(&error_handler);
+
+                async move {
+                    handle_update(update, deps, handler, default_handler, error_handler).await;
+                }
+            })
+            .await;
+    });
+
+    tx
+}
+
+fn spawn_default_worker<Err>(
+    deps: DependencyMap,
+    handler: Arc<UpdateHandler<Err>>,
+    default_handler: Arc<DefaultHandler>,
+    error_handler: Arc<dyn ErrorHandler<Err> + Send + Sync>,
+) -> WorkerTx
+where
+    Err: Send + Sync + 'static,
+{
+    let (tx, rx) = tokio::sync::mpsc::channel(WORKER_QUEUE_SIZE);
+
+    let deps = Arc::new(deps);
+
+    tokio::spawn(async move {
+        ReceiverStream::new(rx)
+            .for_each_concurrent(None, |update| {
```

Comment:
The upper limit of the concurrent tasks here should probably be configureable too.